core/modules/replica/TransactionsApp.cc

```diff
@@ -0,0 +1,189 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/TransactionsApp.h"
+
+// System headers
+#include <chrono>
+#include <iostream>
+#include <stdexcept>
+
+// Qserv headers
+#include "replica/DatabaseServices.h"
+#include "replica/Performance.h"
+#include "replica/ServiceProvider.h"
+#include "util/TablePrinter.h"
+
+using namespace std;
+
+namespace {
+
+string const description =
+    "This application is meant for testing persistent states of super-transaction";
+
+} /// namespace
+
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+TransactionsApp::Ptr TransactionsApp::create(int argc, char* argv[]) {
+    return Ptr(
+        new TransactionsApp(argc, argv)
+    );
+}
+
+
+TransactionsApp::TransactionsApp(int argc, char* argv[])
+    :   Application(
+            argc, argv,
+            ::description,
+            true    /* injectDatabaseOptions */,
+            false   /* boostProtobufVersionCheck */,
+            true    /* enableServiceProvider */
+        ) {
+
+    // Configure the command line parser
+
+    parser().commands(
+        "operation",
+        {   "FIND",
+            "LIST",
+            "BEGIN",
+            "END"
+        },
+        _operation);
+
+    parser().option(
+        "tables-page-size",
+        "The number of rows in the table of a query result set (0 means no pages).",
+        _sqlPageSize);
+
+    parser().flag(
+        "abort",
+        "Abort the transaction rather than normally end it.",
+        _abort);
```

Comment:
Do you need global `abort`, I think it only applies to END command which has its own abort option?

Reply:
Good catch! Eliminated!